### PR TITLE
Initial folder layout to allow further commits

### PR DESCRIPTION
**Pull request by maxgolov**
Initial folder layout to allow further commits. I intentionally left a few changes as `TBD` to allow for additional flexibility:
- coding style
- code ownership
- formatting rules

Different contributing individuals and companies should be able to use their own applicable coding style, i.e. if a module enables integration with a different system or product, it would most likely follow the coding style rules of the product.

I am not sure 100% re. the license, but keeping it Apache License Version 2.0 as the main repo. We may need to add an additional provision that _optional_ community-contributed projects may each be under their own license?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (reyang) [inline .github/ISSUE_TEMPLATE/bug_report.md:7]:
```suggestion
**Describe your environment** Describe any aspect of your environment relevant to the problem, including your platform, build system, version numbers of installed dependencies, etc. If you're reporting a problem with a specific version of a library in this repo, please check whether the problem has been fixed on main branch.
```

File: README.md
Comment (reyang) [inline README.md]: This repo is **not** maintained by .NET maintainers 😄

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (maxgolov) [reply .github/ISSUE_TEMPLATE/bug_report.md:7]: Need to fix this for .NET too :) !

File: README.md
Comment (maxgolov) [reply README.md]: Sorry, already fixed :)

Comment (reyang) [reply README.md]: I think as the 1st step we need to clarify who are the maintainers/approvers.

**Review by reyang — CHANGES_REQUESTED** [review]
Need to figure out who should be the maintainers of this project.

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (reyang) [reply .github/ISSUE_TEMPLATE/bug_report.md]: .NET is correct since it hasn't yet migrated to main branch.

**Comment by maxgolov** [comment]
> Need to figure out who should be the maintainers of this project.

It's not that we have a huge line of maintainers at this point, so I mirrored what was done for .NET : same contributors and maintainers as for the main C++ repository. I'd be glad to be one of the maintainers.

File: README.md
Comment (lalitb) [inline README.md:21]: Probably, we can bring more clarity ( in another PR ) on how the role of approvers and maintainers in PR review. I like the way it is done for `opentelemetry-collector-contrib` (https://github.com/open-telemetry/opentelemetry-collector-contrib):
``
News PRs will be automatically associated with the reviewers based on CODEOWNERS. PRs will be also automatically assigned to one of the maintainers or approvers for facilitation.
The facilitator is responsible for helping the PR author and reviewers to make progress or if progress cannot be made for closing the PR.
``

**Review by lalitb — APPROVED** [review]

**Review by ThomsonTan — APPROVED** [review]

File: .gitattributes
Comment (ThomsonTan) [inline .gitattributes:6]: `.cc` is also a common extension for cpp.

File: README.md
Comment (maxgolov) [reply README.md:21]: Let's do this in a follow-up PR

Comment (reyang) [reply README.md:21]: I think we want to follow what folks are doing here https://github.com/open-telemetry/opentelemetry-dotnet-contrib#support.
I don't think we should assume the opentelemetry/cpp-maintainers and opentelemetry/cpp-approvers are the right group for this contrib repo.

**Comment by reyang** [comment]
> @reyang Can you look on this PR once, as it's blocked on changes requested by you. Thanks :)

I'll approve if we remove https://github.com/orgs/open-telemetry/teams/cpp-maintainers and https://github.com/orgs/open-telemetry/teams/cpp-approvers from the PR. We can either create separate groups (e.g. cpp-contrib-approvers) like what .NET did, or leave it as a TODO for next PR.

File: README.md
Comment (maxgolov) [reply README.md:24]: I removed the notion of `opentelemetry/cpp-maintainers` and `opentelemetry/cpp-approvers`, and added a reference to the `CODEOWNER`S process instead.

**Comment by maxgolov** [comment]
> I'll approve if we remove https://github.com/orgs/open-telemetry/teams/cpp-maintainers and https://github.com/orgs/open-telemetry/teams/cpp-approvers from the PR.

I removed the notion of the main repo maintainers and approvers. I added a reference to the `CODEOWNER`S process, which itself is TBD in a separate PR.

**Review by reyang — APPROVED** [review]
LGTM.

**Pull request merged**
